### PR TITLE
Created TextLinks to replace DisplayLinks and added to announcements

### PR DESCRIPTION
**Pull request by SnowyNate**
TextLinks component has been created to replace the previous function DisplayLinks, which did not support newlines. TextLinks functionality has been added to announcements in Catalog page and AlertDisplay. Fixes issue #233

### CODE REVIEW

File: src/comps/ui/AlertDisplay.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { useEffect, useState } from "react";
 import Alert from "@mui/material/Alert";
-import DisplayLinks from "./DisplayLinks";
+import TextLinks from "./TextLinks";
```
Comment (TheEgghead27) [inline src/comps/ui/AlertDisplay.tsx:3]: I suppose you could have simply kept the component called DisplayLinks to reduce the number of breaking changes, but this is fine

File: src/comps/ui/TextLinks.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const formatTextWithLinks = (text: string) => {
+        const urlRegex = /(https?:\/\/[^\s]+)/g;
+        const parts = text.split(urlRegex);
+
+        return (
+            <>
+                {parts.map((part, index) => {
+                    if (urlRegex.test(part)) {
+                        return (
+                            <a
```
Comment (TheEgghead27) [inline src/comps/ui/TextLinks.tsx:18]: It would be best to use a MUI Link (or other styled link) for UI consistency

**Review by TheEgghead27 — CHANGES_REQUESTED** [review]
This is amazing! I am so sorry I lost track of this PR

I suppose this component should also be used for the socials section of clubs?

**Comment by SnowyNate** [comment]
Updated to PR #262

**Pull request closed without merging**